### ISSUE TITLE
master_event_id key not present in typings for Event class

### ISSUE THREAD

**Issue by ahathaway** [nylas/nylas-nodejs#221]
I am not sure if this qualifies as a bug or a feature request, I suppose an argument could be made for either. Because I am unsure, I will post my suggested fix here and I am also happy to create a PR to introduce the change if the owners of this repository agree.

I have changes ready to PR here: https://github.com/ahathaway/nylas-nodejs

I am currently using the updated version in my integration, would love to see this pulled into the nylas repo so that I can switch back.

Current typings for Event class in file: node_modules/nylas/lib/models/event.d.ts

```typescript
export default class Event extends RestfulModel {
    calendarId?: string;
    iCalUID?: string;
    messageId?: string;
    title?: string;
    description?: string;
    owner?: string;
    participants?: EventParticipant[];
    readOnly?: boolean;
    location?: string;
    when?: {
        start_time?: number;
        end_time?: number;
        time?: number;
        start_date?: string;
        end_date?: string;
        date?: string;
        object?: string;
    };
    busy?: boolean;
    status?: string;
    recurrence?: {
        rrule: string[];
        timezone: string;
    };
    get start(): string | number | undefined;
    set start(val: string | number | undefined);
    get end(): string | number | undefined;
    set end(val: string | number | undefined);
    deleteRequestQueryString(params?: {
        [key: string]: any;
    }): {
        [key: string]: any;
    };
    save(params?: {} | SaveCallback, callback?: SaveCallback): Promise<this>;
    rsvp(status: string, comment: string, callback: (error: Error | null, data?: Event) => void): Promise<this>;
}
```

Unfortunately, when there is a recurrence pattern for an event series, and then the user deletes one of those instances (thus producing an override), Nylas sends a different object over the webhook:

```json
{
  "account_id": "xxxxxxxxxxxxxxxxxxxxxxxxx",
  "busy": true,
  "calendar_id": "xxxxxxxxxxxxxxxxxxxxxxxxx",
  "description": null,
  "ical_uid": "xxxxxxxxxxxxxxxxxxxxxxxxx@google.com",
  "id": "xxxxxxxxxxxxxxxxxxxxxxxxx",
  "location": null,
  "master_event_id": "xxxxxxxxxxxxxxxxxxxxxxxxx",
  "message_id": null,
  "object": "event",
  "original_start_time": 1619460000,
  "owner": " <xxxxxxxxxxxx@xxxxxxxxxxx.com>",
  "participants": [],
  "read_only": false,
  "status": "cancelled",
  "title": "recurring thing with set end date",
  "when": {
    "end_time": 1619463600,
    "object": "timespan",
    "start_time": 1619460000
  }
}
```

Notice the `master_event_id` which indicates that the item is an override event for a series. 

I believe this should be added to the type definition as: `masterEventId?: string;`

Like so: 

```typescript
export default class Event extends RestfulModel {
    calendarId?: string;
    iCalUID?: string;
    messageId?: string;
    title?: string;
    description?: string;
    owner?: string;
    participants?: EventParticipant[];
    readOnly?: boolean;
    location?: string;
    masterEventId?: string;
    when?: {
        start_time?: number;
        end_time?: number;
        time?: number;
        start_date?: string;
        end_date?: string;
        date?: string;
        object?: string;
    };
    busy?: boolean;
    status?: string;
    recurrence?: {
        rrule: string[];
        timezone: string;
    };
    get start(): string | number | undefined;
    set start(val: string | number | undefined);
    get end(): string | number | undefined;
    set end(val: string | number | undefined);
    deleteRequestQueryString(params?: {
        [key: string]: any;
    }): {
        [key: string]: any;
    };
    save(params?: {} | SaveCallback, callback?: SaveCallback): Promise<this>;
    rsvp(status: string, comment: string, callback: (error: Error | null, data?: Event) => void): Promise<this>;
}
```

Thoughts on this?

**Comment by ahathaway** [comment]
I found another key missing: original_start_time/originalStartTime

**Comment by ahathaway** [comment]
Is there any input from the contributing team on this repo? It seems odd that instructions on how to contribute are provided, and then the permissions on this repository are such that the instructions cannot be adhered to (labeling issues for example).

**Comment by mrashed-dev** [comment]
Hey @ahathaway! Thanks for tackling this issue, you are more than welcome to submit a PR! Can you describe what issues you are seeing when attempting to create a PR? I can try to help resolve those issues so we can get the PR started and eventually merged in 😄

**Comment by dereekb** [comment]
Ran into the same issue. Currently manually extending Event right now to access conferencing, originalStartTime and masterEventId, and sending requests with Axios.

https://developer.nylas.com/docs/api/#tag--Events--events-object

(Conferencing isn't available in the above docs, but is available from the API)

Data from the API w/ original values changed:

```
{
        "account_id": "1wvo7bbwtectoslnsppxxxxxx",
        "busy": true,
        "calendar_id": "6sdp3h5r8p0roprwenpxxxxxx",
        "conferencing": {
            "details": {
                "meeting_code": "00000000000",
                "phone": [
                    "+1000000000"
                ],
                "url": "https://zoom.us/j/00000000000"
            },
            "provider": "Zoom Meeting"
        },
        "description": null,
        "ical_uid": "7uoppphl86s6btukjgxxxxxxxx@google.com",
        "id": "675pk7k14azwsao2erxxxxxxx",
        "location": null,
        "master_event_id": "d1jqfyjlgiltadiy1rxxxxxxx",
        "message_id": null,
        "object": "event",
        "original_start_time": 1625508000,
        "owner": " <xxx@test.com>",
        "participants": [
            {
                "comment": null,
                "email": "xxx@test.com",
                "name": null,
                "status": "yes"
            },
            {
                "comment": null,
                "email": "yyy@test.com",
                "name": null,
                "status": "noreply"
            }
        ],
        "read_only": false,
        "reminders": null,
        "status": "confirmed",
        "title": "My Event",
        "when": {
            "end_time": 1625598000,
            "object": "timespan",
            "start_time": 1625594400
        }
    }
```

**Comment by mrashed-dev** [comment]
The values `original_start_time` and `master_event_id` are now available in the Events object, and will be included in the next minor release! 

@dereekb Regarding conferencing, support will be coming very soon - please follow #256 to track the request, thank you!

**Comment by horlaarsco** [comment]
This has been removed again from the types, I check [this](https://github.com/nylas/nylas-nodejs/pull/259) PR you referenced, there was an event.ts but now its events.ts and the masterEventId has been excluded again

**Comment by mrashed-dev** [comment]
Reopening this issue, thanks @horlaarsco we'll take a look as soon as we get a chance!

**Comment by empro-danielkress** [comment]
Same seems to be true for resources[]

**Comment by SubashPradhan** [comment]
Hey all, the masterEventId has been included on v7.6.2: https://github.com/nylas/nylas-nodejs/releases/tag/v7.6.2. We will proceed with closing this issue. Please let us know if you have any concerns.

@empro-danielkress we will fix the resources[] on our next release.
